eosmetrics/emtr-event-recorder.h

```diff
@@ -135,6 +135,27 @@ void               emtr_event_recorder_record_stop_sync   (EmtrEventRecorder *se
                                                            GVariant          *key,
                                                            GVariant          *auxiliary_payload);
 
+EMTR_AVAILABLE_IN_0_4
+void               emtr_event_recorder_start_aggregate_timer (EmtrEventRecorder   *self,
+                                                              const gchar         *event_id,
+                                                              GVariant            *aggregate_key,
+                                                              GVariant            *auxiliary_payload,
+                                                              GCancellable        *cancellable,
+                                                              GAsyncReadyCallback  callback,
+                                                              gpointer             user_data);
+
+EMTR_AVAILABLE_IN_0_4
+EmtrAggregateTimer *emtr_event_recorder_start_aggregate_timer_finish (EmtrEventRecorder  *self,
+                                                                      GAsyncResult       *result,
+                                                                      GError            **error);
+
+EMTR_AVAILABLE_IN_0_4
+EmtrAggregateTimer *emtr_event_recorder_start_aggregate_timer_sync (EmtrEventRecorder  *self,
+                                                                    const gchar        *event_id,
+                                                                    GVariant           *aggregate_key,
+                                                                    GVariant           *auxiliary_payload,
+                                                                    GCancellable       *cancellable,
+                                                                    GError            **error);
```

Comment:
Thinking about the shell case… we can't use the sync API there, certainly not on the UI thread. But the async API will be annoying to use.

Could we do some glue in the library so that there is only a sync API, which returns an EmtrAggregateTimer, which you can call stop() on synchronously. Internally, it would make an async D-Bus call, and if stop() is called before that async call returns, wait for the async call to return before calling stop on the D-Bus object.

Reply:
Done